bricks/dart_frog_prod_server/hooks/pre_gen.dart

```diff
@@ -64,25 +77,30 @@ Future<void> preGen(
     },
   );
 
-  await reportExternalPathDependencies(
-    projectDirectory,
-    onViolationStart: () {
-      context.logger
-        ..err('All path dependencies must be within the project.')
-        ..err('External path dependencies detected:');
-    },
-    onExternalPathDependency: (dependencyName, dependencyPath) {
-      context.logger.err('  \u{2022} $dependencyName from $dependencyPath');
-    },
-    onViolationEnd: () {
-      exit(1);
-    },
-  );
-
   final customDockerFile = io.File(
     path.join(projectDirectory.path, 'Dockerfile'),
   );
 
+  // Get all the internal path packages
+  final internalPathDependencies = (await getPathDependencies(projectDirectory))
+      .where(
+        (dependencyPath) =>
+            path.isWithin(projectDirectory.path, dependencyPath),
+      )
+      .toList();
+
+  // Then create the external packages folder
+  // and add it to the list of path packages.
+  final externalDependencies = await createExternalPackagesFolder(
+    projectDirectory,
+    copyPath: copyPath,
+  );
```

Comment:
**Nit**: I personally think the code is already self-explanatory and it is doing a fantastic job at describing what is being done. Unlike code, comments do not get updated as code changes and thus they might not always tell the truth about the code. Hence, if the code is already clear I would personally disregard the comments. 

These thoughts are not mine, they have been stolen from Uncle's Bob book, I think this [document](https://bpoplauschi.github.io/2021/01/20/Clean-Code-Comments-by-Uncle-Bob-part-2.html) cites some of it.

```suggestion
  final internalPathDependencies = (await getPathDependencies(projectDirectory))
      .where(
        (dependencyPath) =>
            path.isWithin(projectDirectory.path, dependencyPath),
      )
      .toList();

  final externalDependencies = await createExternalPackagesFolder(
    projectDirectory,
    copyPath: copyPath,
  );
```